src/inmanta/ast/statements/generator.py

```diff
@@ -427,10 +741,15 @@ def resume(self, requires: Dict[object, VariableABC], resolver: Resolver, queue:
             subexpression: ExpressionStatement = (
                 self.expression.if_expression if self.condition_value else self.expression.else_expression
             )
+            subexpression_requires: dict[object, VariableABC[object]] = (
+                subexpression.requires_emit(resolver, queue)
+                if self.lhs is None
+                else subexpression.requires_emit_gradual(resolver, queue, self.lhs)
```

Comment:
Why have this else everywhere instead of making requires_emit_gradual shortcut to requires_emit when the lhs parameter is None?